rewrite-java/src/main/java/org/openrewrite/java/cleanup/RemoveCallsToSystemGc.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2022 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.cleanup;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.tree.J;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Set;
+
+public class RemoveCallsToSystemGc extends Recipe {
+    @Override
+    public String getDisplayName() {
+        return "Remove calls to `System.gc()` method";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Removes calls to System.gc()";
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return Collections.singleton("RSPEC-1215");
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(1);
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getSingleSourceApplicableTest() {
+        return new UsesMethod<>("java.lang.System gc()");
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+            @Override
+            public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext context) {
+                J.MethodInvocation invocation = super.visitMethodInvocation(method, context);
+                if (new MethodMatcher("java.lang.System gc()").matches(invocation)) {
+                    return null;
```

Comment:
```suggestion
                    doAfterVisit(new EmptyBlock());
                    return null;
```
as per @tkvangorder [in this comment](https://github.com/openrewrite/rewrite/issues/1664#issuecomment-1101563778)

Reply:
I might be wrong; just hoping to be helpful before the OpenRewrite team themselves chime in; typically from a different timezone. :)

Reply:
Thank you. Done in https://github.com/openrewrite/rewrite/pull/2309/commits/3305542f3b52344944a00ebb8f09f6c5f3b9a65d

Reply:
Nicely done, and good work on expanding the test to cover the empty block removal. Now let's hope it passes a proper review as well.